### ISSUE TITLE
Segmentation fault after creating a copy of a gsTHBSplineBasis.

### ISSUE THREAD

**Issue by sofimp23** [gismo/gismo#346]
Dear all,
the following code:
```cpp
#include <gismo.h>

using namespace gismo;

void printSize(const gsTHBSplineBasis<2, real_t>& basis)
{
     gsTHBSplineBasis<2> copy;
     copy = basis;
     gsInfo << "Size of the copy: " << copy.size() << std::endl;     
}

int main()
{    
    gsKnotVector<> kv(0, 1.0, 3, 3);
    gsTensorBSplineBasis<2> b(kv, kv);
    gsTHBSplineBasis<2, real_t> basis(b);
    
    printSize(basis);

    return 0;
}
```
gives a "Segmentation fault." just before it terminates.

Our understanding (with @dominik-mokris-mtu) is that once `copy` goes out of scope (upon returning from `printSize`), a destructor of `gsHTensorBasis` calls `freeAll`, which deletes all the underlying tensor-product bases. At the end of the program, the same destructor is called once again on `basis`, trying to delete the underlying bases for the second time.

**Question 1** Is this behaviour intended?

**Question 2** If yes, how can one create a copy of `gsTHBSplineBasis` safely, i.e., without hurting the original when the copy goes out of scope?

**Comment by codewerfer** [comment]
```cpp
#include <gismo.h>

using namespace gismo;

void printSize(const gsTHBSplineBasis<2, real_t>& basis)
{
    gsTHBSplineBasis<2>::uPtr copy;
    copy = basis.clone();
    gsInfo << "Size of the copy: " << copy->size() << std::endl;
}

int main()
{
    gsKnotVector<> kv(0, 1.0, 3, 3);
    gsTensorBSplineBasis<2> b(kv, kv);
    gsTHBSplineBasis<2, real_t> basis(b);

    printSize(basis);

    return 0;
}
```

**Comment by filiatra** [comment]
it's actually a bug, since  copy assignment operator should be defined here..

**Comment by dominik-mokris-mtu** [comment]
Good! Shall I write the copy assignment operator or do you want to do it?

**Comment by filiatra** [comment]
Fixed in f90c3636ddf736623b1b575d6ce3377489a1f52a
